src/plat/bcm2711/config.cmake

```diff
@@ -26,6 +26,25 @@ if(KernelPlatformRpi4)
     list(APPEND KernelDTSList "src/plat/bcm2711/overlay-rpi4.dts")
     list(APPEND KernelDTSList "src/plat/bcm2711/overlay-rpi4-address-mapping.dts")
 
+    if(NOT DEFINED RPI4_MEMORY)
+        # By default we assume an RPi4B model with 8GB of RAM
+        set("${RPI4_MEMORY}" "8192")
+    endif()
+
+    if("${RPI4_MEMORY}" STREQUAL "1024")
+        list(APPEND KernelDTSList "src/plat/bcm2711/overlay-rpi4-1gb.dts")
+    elseif("${RPI4_MEMORY}" STREQUAL "2048")
+        list(APPEND KernelDTSList "src/plat/bcm2711/overlay-rpi4-2gb.dts")
+    elseif("${RPI4_MEMORY}" STREQUAL "4096")
+        list(APPEND KernelDTSList "src/plat/bcm2711/overlay-rpi4-4gb.dts")
```

Comment:
Should check that this is only possible when 64-bit ARM is being used.

Reply:
Might be better/easier to merge this and adjust https://github.com/seL4/seL4/pull/976 or the other way around.